src/app/pages/send/send-crypto-asset-form/components/recipient-fields/generic-recipient-field.tsx

```diff
@@ -0,0 +1,62 @@
+import { TextInputFieldError } from '@app/components/field-error';
+import { RecipientFieldType } from '@app/pages/send/send-crypto-asset-form/components/recipient-type-dropdown/recipient-type-dropdown';
+
+import { SelectAccountButton } from '../recipient-accounts-drawer/select-account-button';
+import { RecipientField } from '../recipient-field';
+import { RecipientDropdownOverlay } from '../recipient-type-dropdown/components/recipient-dropdown-overlay';
+import { useRecipientSelectFields } from './hooks/use-recipient-select-fields';
+import { RecipientFieldBnsName } from './recipient-field-bns-name';
+
+interface GenericRecipientFieldProps {
+  bnsFn(client: any, name: string, isTestnet?: boolean): Promise<string | null>;
+}
+export function GenericRecipientField({ bnsFn }: GenericRecipientFieldProps) {
+  const {
+    isSelectVisible,
+    showRecipientAccountsModal,
+    onSelectRecipientFieldType,
+    onSetIsSelectVisible,
+    selectedRecipientField,
+  } = useRecipientSelectFields();
+
+  const recipientDropdown = (
```

Comment:
At some point we should swap this out for the new `Select`? Hopefully we can replace these custom built ones at some point.